tmpl/wiki/tutorial-lwt.md

```diff
@@ -74,508 +111,327 @@ Assuming you are building for POSIX using native kernel sockets, compile the app
 ```
 
 If you are building for a different MirageOS platform, change the `--unix` switches appropriately (presently either `--xen` or `--unix`).
-The `OS` module packs several sub-modules depending of the backend, [unix](https://github.com/mirage/mirage-platform/tree/master/unix/lib)
+The `OS` module packs several sub-modules depending on the backend, [unix](https://github.com/mirage/mirage-platform/tree/master/unix/lib)
 or [xen](https://github.com/mirage/mirage-platform/tree/master/xen/lib).
 
 ###Solution
 
 ```
   open V1_LWT (* provides the CONSOLE signature *)
-  open Lwt    (* provides bind and join *)
+  open Lwt    (* provides >>= and join *)
   open OS     (* provides Env, Main, and Time *)
 
   module Main (C : CONSOLE) = struct
     let start c =
-      bind (join [
-        bind (Time.sleep 1.0) (fun () ->
-          C.log c "Heads"; return ()
-        );
-        bind (Time.sleep 2.0) (fun () ->
-          C.log c "Tails"; return ()
-        );
-      ]) (fun () ->
-        C.log c "Finished"; return ()
-      )
+      Lwt.join [
+        (Time.sleep 1.0 >>= fun () -> C.log_s c "Heads");
+        (Time.sleep 2.0 >>= fun () -> C.log_s c "Tails")
+      ] >>= fun () ->
+      C.log_s c ("Finished")
   end
 ```
 
 This code is also found in `lwt/src/unikernels.ml` in the `mirage-skeleton` code repository. Build it by setting the TARGET environment variable to `heads1` before running `mirage configure`.
 
-###Syntax Extensions
 
-Using Lwt does sometimes require significantly restructuring code, and in particular doesn't work with many of the more imperative OCaml control structures such as `for` and `while`. Luckily, Lwt includes a comprehensive [pa_lwt](http://ocsigen.org/lwt/api/Pa_lwt) syntax extension that makes writing threaded code as convenient as vanilla OCaml. MirageOS includes this extension by default, so you can use it anywhere you want.
+##Challenge 2: Echo server
```

Comment:
Now that the echo server isn't demonstrating the syntax extensions, it's not clear to me what it's teaching.  Perhaps it could be removed entirely, or used to demonstrate a different kind of composition?